### ISSUE TITLE
Feature request: Provide access to Guzzle client to assist with debugging

### ISSUE THREAD

**Issue by patricknelson** [noweh/twitter-api-v2-php#42]
It would be awesome if we could get access to the Guzzle client for debugging or other purposes. This would make it easier for example to get details about what happened after requests (both normal and error scenarios).

For example, Twitter ("X") includes their rate limiting details in headers such as `x-rate-limit-remaining` which don't appear to be easily available yet in this library. See: https://developer.twitter.com/en/docs/twitter-api/rate-limits#headers-and-codes

**Comment by joedolson** [comment]
You can do a try/catch with your query, e.g.:

```
try {
	$return    = $client->tweet()->create()->performRequest( $status );
} catch ( Exception $e ) {
	// Get Guzzle exception response.
	$response = $e->getResponse();
	$body       = $response->getBody()->getContents();
	$headers  = $response->getHeaders();
}
```

The headers will give you access to the rate limit data.

**Comment by patricknelson** [comment]
I actually stumbled upon that later on. However, that requires an exception to actually occur (perhaps by actually surpassing a rate limit). The problem is that my use case was to access these headers in order to keep track of it ahead of time _before_ hitting any limits, mainly as a warning that I could surface to the end user like in a dashboard somewhere.

After further testing, I found that I couldn't use this library anyway. I'll keep this here though in case anyone else still might want that functionality _without_ having to use exceptions to gain access to those headers. 🙂
